app/src/main/java/net/inkyquill/pocketeditor/sync/SyncEngine.kt

```
@@ -430,15 +433,51 @@ class SyncEngine internal constructor(
+        }
+        val localManifestSha = sha256(BookManifest.encode(localManifest).encodeToByteArray())
         val manifestEntry = entries[MANIFEST_PATH]
-        val manifestOutbox = pending[MANIFEST_PATH]
+        var manifestOutbox = pending[MANIFEST_PATH]
+        if (manifestOutbox == null && MANIFEST_PATH in pendingPublications) {
+            val metadataBase = metadata.mergeBase(bookId, MANIFEST_PATH)
+            val durableBase = baseStore.read(bookId, MANIFEST_PATH)
+            if (
+                metadataBase != null && durableBase != null &&
+                metadataBase.sha256 == durableBase.sha256 &&
+                metadataBase.remoteRevision == durableBase.remoteRevision
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)